app/tests/timecontrol_test.cpp

```diff
@@ -0,0 +1,59 @@
+#include <time/timecontrol.hpp>
+
+#include "doctest/doctest.hpp"
+
+using namespace fastchess;
+
+TEST_SUITE("TimeControl") {
+    TEST_CASE("moves/time+increment") {
+        TimeControl::Limits limits;
+        limits.moves = 3;
+        limits.time = 10000;
+        limits.increment = 100;
+        limits.timemargin = 100;
+
+        TimeControl tc(limits);
+
+        CHECK(tc.updateTime(5555) == true);
+        CHECK(tc.getTimeLeft() == 4645);
```

Comment:
instead of hardcoding the value here you can instead write the expression which calculates this, like the time + increment - 5555 + increment, makes it easier to follow

Reply:
ahh ok

Reply:
fixed 👍